tree/criterion_interface.hpp

```diff
@@ -0,0 +1,70 @@
+#ifndef MACHINELEARNING_TREE_CRITERION_HPP
+#define MACHINELEARNING_TREE_CRITERION_HPP
+
+#include <math.h>
+
+namespace tree {
+
+/**
+ * @brief Base class that defines the many criterious of splitting.
```

Comment:
Use latex-like slash for the comments attributes instead of @. Like it was done in the other files.